anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -2162,6 +2168,24 @@ def add_or_update_image(
                                 manifest = json.dumps({})
                             if not parent_manifest:
                                 parent_manifest = json.dumps({})
+
+                            # If image status is already analyzed, generate a UserAnalyzeImageCompleted event
+                            if image_record[
+                                "analysis_status"
+                            ] == anchore_engine.subsys.taskstate.complete_state(
+                                "analyze"
+                            ):
+                                for new_id in new_image_detail:
```

Comment:
This may not be accurate. We should only be generating events for the tags in this request, not all tags. Otherwise the same event will be generated for a tag multiple times. The loop should probably iterate over `tags` instead